src/os/bluestore/BlueStore.cc

```diff
@@ -5916,6 +5916,8 @@ int BlueStore::_do_read(
             return r;
           return 0;
 	});
+	if(r<0)
```

Comment:
i believe we also expect r to be 0 here, too.  can you change this to an assert and run through ceph_objectstore_tool --gtest_filter=*/2 to ensure the tests pass?  Thanks!